app/controllers/mixins/generic_show_mixin.rb

```diff
@@ -1,6 +1,7 @@
 module Mixins
   module GenericShowMixin
     def show
+      return if perfmenu_click?
```

Comment:
I wonder how generic this line is and if it should be in the mixin.

Also I wonder why it's needed. Is it because the when the perf menu is clicked the handler ends up calling show? And is that necessary?

I hope you get what I mean. This seems like the kind of spaghetti that we should remove rather that something that should be included in the generic mixin. But one cannot be sure. Can you take a look into that?